extensions/pkg/controller/common/context.go

```diff
@@ -81,10 +67,12 @@ type RESTConfigContext struct {
 	restConfig *rest.Config
 }
 
-// InjectConfig injects the given REST config into the context.
-func (cc *RESTConfigContext) InjectConfig(config *rest.Config) error {
-	cc.restConfig = config
-	return nil
+// NewRESTConfigContext creates a ClientContext with the REST config.
+func NewRESTConfigContext(mgr manager.Manager) RESTConfigContext {
```

Comment:
TBH, I don't see much value left in this package, now that things are not injected anymore.
I vote for dropping it as well and simply pass the manager around like everywhere else in this PR :)